sklearn/manifold/_locally_linear.py

```diff
@@ -222,7 +222,7 @@ def _locally_linear_embedding(
         raise ValueError(
             "output dimension must be less than or equal to input dimension"
         )
-    if n_neighbors >= N:
+    if n_neighbors > N:
```

Comment:
I actually think since we're passing `n_neighbors + 1` to `NearestNeighbors` here a few lines above,

https://github.com/scikit-learn/scikit-learn/pull/29716/files#diff-fda62beef1d843550673f77d3e2c36344a4a9ea5fb1d1dc2b9f1f0d23405522bR215

the condition is correct, and the error message is wrong, i.e. here `n_neighbors` does NOT include the data point itself. (I find it quite confusing though)

cc @adam2392 maybe

Reply:
If the condition is incorrect, it seems like an in-line comment should be added, as I agree it's confusing. 

Unfortunately, we can't just do `n_neighbors += 1`, as it's used below. 

Why are we passing in `n_neighbors + 1` into `NearestNeighbors` anyways? Def something worth clarifying in the code + then fix the error message.

Reply:
I believe that since n_neighbors is increased by 1 only in the NearestNeighbors function, it does not affect the original n_neighbors variable itself. Therefore, when the validation checks the value of n_neighbors, it will only check the original variable, not the incremented one used internally by NearestNeighbors.

The reason for incrementing n_neighbors by 1 in NearestNeighbors is to ensure that the point itself is included in the list of neighbors. This is because most nearest neighbors algorithms will include the point being queried as the first neighbor. By requesting n_neighbors + 1 neighbors, the algorithm guarantees that the actual n_neighbors closest points, excluding the query point itself, are returned.

Reply:
that tells me the condition is correct, but the message is wrong.